gcloud/datastore/_implicit_environ.py

```diff
@@ -96,4 +156,102 @@ def get_default_dataset_id():
     :rtype: string or ``NoneType``
     :returns: The default dataset ID if one has been set.
     """
-    return DATASET_ID
+    return _DEFAULTS.dataset_id
+
+
+def get_connection():
+    """Shortcut method to establish a connection to the Cloud Datastore.
+
+    Use this if you are going to access several datasets
+    with the same set of credentials (unlikely):
+
+    >>> from gcloud import datastore
+
+    >>> connection = datastore.get_connection()
+    >>> key1 = datastore.Key('Kind', 1234, dataset_id='dataset1')
+    >>> key2 = datastore.Key('Kind', 1234, dataset_id='dataset2')
+    >>> entity1 = datastore.get(key1, connection=connection)
+    >>> entity2 = datastore.get(key2, connection=connection)
+
+    :rtype: :class:`gcloud.datastore.connection.Connection`
+    :returns: A connection defined with the proper credentials.
+    """
+    implicit_credentials = credentials.get_credentials()
+    scoped_credentials = implicit_credentials.create_scoped(SCOPE)
+    return Connection(credentials=scoped_credentials)
+
+
+def _lazy_connection():
+    """Alias wrapper for ``get_connection``.
+
+    Unit tests need to be able to replace :func:`get_connection`
+    so we can't wrap the actual ``function`` object in a ``_LazyProperty``.
+    """
+    return get_connection()
+
+
+def set_default_connection(connection=None):
+    """Set default connection either explicitly or implicitly as fall-back.
+
+    :type connection: :class:`gcloud.datastore.connection.Connection`
+    :param connection: A connection provided to be the default.
+    """
+    connection = connection or get_connection()
+    _DEFAULTS.connection = connection
+
+
+def get_default_connection():
+    """Get default connection.
+
+    :rtype: :class:`gcloud.datastore.connection.Connection` or ``NoneType``
+    :returns: The default connection if one has been set.
+    """
+    return _DEFAULTS.connection
+
+
+class _LazyProperty(object):
+    """Descriptor for lazy loaded property.
+
+    This follows the reify pattern: lazy evaluation and then replacement
+    after evaluation.
+
+    :type name: string
+    :param name: The name of the attribute / property being evaluated.
+
+    :type method: callable that takes no arguments
+    :param method: The method used to evaluate the property.
```

Comment:
`_method` seems misleading here:  for instance, `_lazy_dataset_id` is a free function, not a method.  Maybe `_deferred` or `_deferred_callable` instead?